### DISCUSSION TITLE
Plotting the way forward for {admiralmetabolic}

### DISCUSSION THREAD

**Discussion by manciniedoardo** [pharmaverse/admiralmetabolic#5] [General]
## Please use this space to share any thoughts/comments/opinions regarding what we should focus on for our first {admiralmetabolic} release. 

If you have experience in the metabolism/obesity TA, these leading questions could help guide your thought:

- Are there any specific ADaM datasets that your studies often provide, which are unique to this TA? 
  - If so, what are they and why is it necessary to create them? 
  - If not (or additionally) do you perhaps need to add any specific variables to your subject-level, BDS or occurrence ADaMs? 
  - Similarly
- Is there any TA-specific metadata that drives ADaM creation? Scales, mappings, etc...? 
  - If yes, how do you approach using it during your ADaM creation?

The tools we provide in {admiralmetabolic} will hopefully aim to address any answers to the questions above. 

The {admiralmetabolic} package will also provide test data to play around with said tools and generate ADaMs pertinent to the TA. Is there any TA-specific test data that we should create, over and above what is in [pharmaversesdtm](https://pharmaverse.github.io/pharmaversesdtm/)? This could include completely new datasets or edited versions of existing ones (e.g. to introduce obese patients).
  
  We can use your comments on the above to inform our way forward 😄

**Comment by AndersAskeland** [comment]
From the top of my head, I'm thinking that oral glucose tolerance test (OGTT) and continuous glucose monitoring (CGM) might be relevant in relation to diabetes, and diet registration and physical activity can be relevant in the case of obesity.

**Comment by manciniedoardo** [comment]
## Obesity

Attendees to the 2024-07-03 meeting shared that when working in this indication, SDTM datasets that are relevant are:
* `VS` (with obese patients)
* `LB`
* `QS`
  
We'd be looking to create test data for these to run our examples/templates/tests. Even if these datasets are already existing in `{pharmaversesdtm}`, we'd likely need to make our own versions of them (e.g. to add obese patients, or to add specific tests that are collected in obesity trials, etc.).

Correspondingly, the following ADaM datasets could be relevant for `{admiralmetbaolic}`
* `ADVS`
* `ADLB` 
* `ADQS`

We'd be looking to create templates and vignettes showcasing how to build the datasets, and showcasing any `{admiralmetabolic}` functionality that we build.

**Reply by manciniedoardo** [reply]
## Questionnaires  (`QS`)

* According to @Sonali-netizen often questionnaire ADaMs are subdivided into ADaMs for different questionnaires (e.g. `ADEQ5D`) due to complexities involving derived parameters/computed scores.
* Potential questionnaires of interest: EQ5D, work-limiting questionnaire, SF36, CoEQ, UWQoL, PHQ9, TFEQ, CSSRS (@Mikkel is that all)?
* We'd need to investigate which of these are proprietary as that would inform what sort of approach we can take, e.g. what test data we make. Also, how to deal with different versions of the quesdtionnaires? @Sonali-netizen can you look into this?

**Reply by manciniedoardo** [reply]
## Labs (`LB`)

* This is the most complicated one according to Piotr as it requires custom functions, imputations and specific endpoints (Piotr can you expand)?
* Differences between Roche and NN approach for lab test units - NN favours an excel file with unit conversions while Roche favours multiple parameters with different units.
* Fasting parameters are of interest - do we have that in the existing `LB` data in `{pharmaversesdtm}`? If not, should we add?.

**Reply by manciniedoardo** [reply]
## Miscellaneous

- BMI is often rederived in ADaMs, in our documentation we may want to indicate how to do that using `{admiral}` functionality
- Estimated Glomerular Filtration Rate  (EGFR) is often derived  (@ Piotr - in which ADaM?) - could this be an opportunity to write a wrapper for `derive_param_computed()`, [like is done for BSA](https://pharmaverse.github.io/admiral/reference/derive_param_bsa.html?q=derive_param_bsa#null)?
- Not many existing published standards

**Reply by AndersAskeland** [reply]
> ## Questionnaires (`QS`)
> * According to @Sonali-netizen often questionnaire ADaMs are subdivided into ADaMs for different questionnaires (e.g. `ADEQ5D`) due to complexities involving derived parameters/computed scores.
> * Potential questionnaires of interest: EQ5D, work-limiting questionnaire, SF36, CoEQ, UWQoL, PHQ9, TFEQ, CSSRS (@mikkel is that all)?
> * We'd need to investigate which of these are proprietary as that would inform what sort of approach we can take, e.g. what test data we make. Also, how to deal with different versions of the quesdtionnaires? @Sonali-netizen can you look into this?

While I am unsure how previous extensions have worked, I am wondering if we should create stuff based general questionnaires in admiralmetabolic (i.e. questionnaires used for many types of clinical trials). For instance, I am unsure if SF-36 (Short Form Health Survey), EQ-5D (EuroQol-5 Dimensions), and PHQ-9 (patient Health Questionnaire-9) might be too general, and should perhaps be reserved for Admiral, or a different extension at a later time?

On the other side, UWQoL (Impact of Weight on Quality of Life) and TFEQ (Three-Factor Eating Questionnaire) seems more relevant. What does everyone else think?

**Reply by manciniedoardo** [reply]
@AndersAskeland 

> While I am unsure how previous extensions have worked, I am wondering if we should create stuff based general questionnaires in admiralmetabolic (i.e. questionnaires used for many types of clinical trials). For instance, I am unsure if SF-36 (Short Form Health Survey), EQ-5D (EuroQol-5 Dimensions), and PHQ-9 (patient Health Questionnaire-9) might be too general, and should perhaps be reserved for Admiral, or a different extension at a later time?
> 
> On the other side, UWQoL (Impact of Weight on Quality of Life) and TFEQ (Three-Factor Eating Questionnaire) seems more relevant. What does everyone else think?

I think that makes a lot of sense. I guess the two questions are:
- are those questionnaires open to use?
- do you need to do anything special on the ADaM side for them? (Because if not, I would question the need for specific functionality for them).

**Comment by manciniedoardo** [comment]
## Diabetes

Attendees to the 2024-07-03 meeting discussed the following:

- Could be a future focus area, but no team members / sponsor companies are currently invested here so why target it?
- Possibility here to look at area under the curve calculations - this is similar to what is done with PK data, so could we look at existing admiral templates? [See here](https://pharmaverse.github.io/admiral/articles/pk_adnca.html) for examples.
- There could be some published standards here - e.g. see [here](https://www.cdisc.org/standards/therapeutic-areas/diabetes-type-1-pediatrics-and-devices/diabetes-type-1-therapeutic-area)

**Comment by AndersAskeland** [comment]
I have been looking up publicly available protocols previously used for obesity trials, and from that found that some common endpoints used are:

- Change in weight / BMI (I would expect the BMI [`derive_param_bmi()`](https://pharmaverse.github.io/admiral/reference/derive_param_bmi.html) function in ADMIRAL would be sufficient for this). 
- Pre-defined set reductions of body weight / BMI. For instance, participants that have achieved a > 5% weight loss reduction. 
- Waist circumference. This would not require anything, however, I know that in academia, there has started to be some focus on waist / hip ratio as an alternative to BMI. This could perhaps be a potential derivation to be used?
- Physical functioning score (SF-36)
- Physical function domain (5-items) score (IWQoL-Lite for CT
- Body composition (as assessed by Dual Energy X-ray Absorpmetry (DEXA)
   - Total fat mass
   - Lean body mass
   - Visceral fat mass

In relation to DXA / DEXA, I am unsure if anything if any functions are needed, but I could imagine a vingette being useful. I will send the links to some of the protocols in our teams channel.

**Reply by manciniedoardo** [reply]
Thanks @AndersAskeland for this. I also had a quick scan for recent obesity trials on ct.gov and encountered much the same as you. My thoughts:

> * Change in weight / BMI (I would expect the BMI [`derive_param_bmi()`](https://pharmaverse.github.io/admiral/reference/derive_param_bmi.html) function in ADMIRAL would be sufficient for this).

Agree here, debatable whether we need to include anything.

> * Pre-defined set reductions of body weight / BMI. For instance, participants that have achieved a > 5% weight loss reduction.

I noticed this quite a lot. As such I would recommend a small vignette showcasing the use of admiral to create criterion flags in ADVS for this maybe? How is this usually done at NN? Worth noting there is a brand new criterion-flag function added to admiral, so we are starting at a right time.

> * Waist circumference. This would not require anything, however, I know that in academia, there has started to be some focus on waist / hip ratio as an alternative to BMI. This could perhaps be a potential derivation to be used?

yes agree - i've seen hip circumference mentioned a lot also. where is this usually mapped in SDTM? could we add to our test data?

> * Body composition (as assessed by Dual Energy X-ray Absorpmetry (DEXA)
>   
>   * Total fat mass
>   * Lean body mass
>   * Visceral fat mass
> 
> In relation to DXA / DEXA, I am unsure if anything if any functions are needed, but I could imagine a vingette being useful. I will send the links to some of the protocols in our teams channel.

Thanks. I don't know much about this myself, but i did notice after some googling that for lean body mass there is something called estimated Lean Body Mass (eLBM). If this is of interest, it looks like it can be calculated [using a formula using weight and height](https://www.calculator.net/lean-body-mass-calculator.html) - might be a nice easy function to implement.  

Would you see the need for some test data here as well?

**Comment by biocyberman** [comment]
@manciniedoardo As it would sound strange at first on clinging to Excel, but it will be much easier to work on specifying data structure with an Excel/Google Worksheet. We can also extract the structure easier in the future. 

I also think it will be easier to put use cases on a wiki page here on the github repo.

**Reply by manciniedoardo** [reply]
Sorry @biocyberman I'm not following - what data structure are you talking about with respect to Excel?
